Wiki: Further clarify page for building alvr

some-username-here1 opened the pull request:
> some-username-here1/ALVR@db764b75db1bd048203ae0787d909edb03943c74: Add a few missing dependency packages for Ubuntu (Ubuntu renamed to "Debian / Ubuntu / Pop_OS") under "Server build"; Add subsection under "Client build" for installing android developer packages for various Linux flavors from upstream and adding Linux directory examples for environment variables.

The different directories listed for Arch and Gentoo for the `ANDROID_SDK_ROOT` environment variable is based off of each wiki's respective pages, [1](https://wiki.archlinux.org/title/Android) and [2](https://wiki.gentoo.org/wiki/Android_studio), so apologies if I got the directory wrong. The packages for all mentioned flavors were also looked up beforehand, comparing to and from, but if anyone who has those flavors would like to make changes or fixes to the pull, feel free!

zarik5 commented on wiki/Building-From-Source.md:
```diff
@@ -8,6 +8,21 @@ You need to install [rustup](https://www.rust-lang.org/tools/install).
 
 On Windows you need also [Chocolatey](https://chocolatey.org/install).
 
+# Users with Nvidia GPUs
+
+Before building the server, those with nvidia GPUs would have to build and install [`FFmpeg/nv-codec-headers`](https://github.com/FFmpeg/nv-codec-headers). The nv-codec-headers requires at least the following driver version for the respective OS to work properly, taken from nv-codec-header's README:
```
Please specify this is for Linux only. Also people on AMD that omit `--no-nvidia` need to install this, so it's better to make it a requirement regardless.

some-username-here1 replied:
How's this?

> Before building the server, those on Linux with would have to build and install [`FFmpeg/nv-codec-headers`](https://github.com/FFmpeg/nv-codec-headers). The nv-codec-headers, for nvidia users at least, requires driver version `520.56.06` or later to work properly; Taken from nv-codec-header's README.

m00nwtchr replied:
Yeah that looks fine

zarik5 commented on wiki/Building-From-Source.md:
```diff
@@ -44,17 +59,17 @@ cargo xtask prepare-deps --platform [your platform] [--gpl] [--no-nvidia]
-* This is not required if your Linux distribution includes Vulkan support in its ffmpeg package, like Arch (see above).
+* **Windows only:** Use the `--gpl` flag if you want to download, build and bundle FFmpeg inside the ALVR server. Keep in mind that on Windows this is needed only for software encoding. As the name suggest, if you use this flag you can only redistribute the final package as GPLv2.0 licensed (because of the x264 encoder).
+* Use the flag `--no-nvidia` only on Linux and if you have a AMD gpu.
+* This command is not required if your Linux distribution includes Vulkan support in its ffmpeg package, like Arch (see above).
```
As of the latest changes, this command is always required, so delete this line.

some-username-here1 replied:
The "Vulkan support in its ffmpeg package" line?

zarik5 replied:
Yes, because now Vulkan is always built from source and statically linked.

some-username-here1 replied:
Done!

zarik5 commented on wiki/Building-From-Source.md:
````diff
@@ -65,14 +80,41 @@ If you want to edit and rebuild the code, you can skip the `prepare-deps` comman
+* Latest Android SDK Platform-Tools 33 (Android 13)
+* Latest Android NDK (currently v25.1.8937393)
+
+On Linux, the three mentioned developer applications can be installed on Linux from upstream.
+
+* **Arch**
+
+  ```bash
+  sudo pacman -S tandroid-sdk-platform-tools sdkmanager android-ndk
+  ```
+
+* **Gentoo**
+
+  * `dev-util/android-studio`
+  * `dev-util/android-sdk-update-manager`
+  * `dev-util/android-ndk >= 25.1`
+
+* **Debian 12 / Ubuntu 22.10 / Pop!\_OS 22.10** (Debian requires to have the `non-free` repository enabled)
+  ```bash
+  sudo apt install android-sdk-platform-tools-common sdkmanager google-android-ndk-r25b-installer
+  ```
````
There have been reports that the Arch package doesn't work. I personally prefer leaving the SDK/NDK installation instructions as unspecified, since it varies greatly from distro to distro. Instructions can be easily found online.

some-username-here1 replied:
Would it better to link to their wiki page, then? for all three distros, or just Arch? Gentoo too?

m00nwtchr replied:
IIRC using packages is more difficult because then the sdkmanager (and by extension the build system) can't automatically install the required SDK components, so you need to make sure to have them manually (either by installing everything from the repos or running sdkmanager as root)

m00nwtchr replied:
I think ideally we should just remove the old client asap and just switch to the one which doesn't require android studio stuff to build

some-username-here1 replied:
So it'd be best to leave out the mention of these packages?

zarik5 commented on wiki/Building-From-Source.md:
````diff
@@ -65,14 +80,41 @@ If you want to edit and rebuild the code, you can skip the `prepare-deps` comman
+  sudo apt install android-sdk-platform-tools-common sdkmanager google-android-ndk-r25b-installer
+  ```
+
+Set the environment variables:
+  * `JAVA_HOME`:
+    * Windows: `C:\Program Files\Android\Android Studio\jre`
+    * Linux: `/usr/lib/jvm/default-java/bin`
+  * `ANDROID_SDK_ROOT`:
+    * Windows: `%LOCALAPPDATA%\Android\Sdk`
+    * Linux:
+      * Debian / Ubuntu / Pop!\_OS : `~/AndroidSDK`
+      * Arch: `~/Android/Sdk`
+      * Gentoo: `~/Android`
+  * `ANDROID_NDK_HOME`:
+    * Windows: `%LOCALAPPDATA%\Android\Sdk\ndk\25.1.8937393`
+    * Linux: `/usr/lib/android-sdk/ndk`
````
Again, I would leave out Linux examples since they vary depending on the installation method.

some-username-here1 replied:
What about mention of the directory differing from install to install? With moving linux mentions to their own section, such as:

> For Windows, set the environment variables:
> 
>   * `JAVA_HOME`:
>     * `C:\Program Files\Android\Android Studio\jre`
>   * `ANDROID_SDK_ROOT`:
>     * `%LOCALAPPDATA%\Android\Sdk`
>   * `ANDROID_NDK_HOME`:
>     * `%LOCALAPPDATA%\Android\Sdk\ndk\25.1.8937393`
> 
> For Linux, the directories for the environment variables can greatly differ from the type of install to where it's installed, but generally they would be as:
> 
>   * `JAVA_HOME`:
>     * `/usr/lib/jvm/default-java/bin`
>   * `ANDROID_SDK_ROOT`:
>     * Debian / Ubuntu / Pop!\_OS: `~/AndroidSDK`
>     * Arch: `~/Android/Sdk`
>     * Gentoo: `~/Android`
>   * `ANDROID_NDK_HOME`:
>     * `/usr/lib/android-sdk/ndk`

Or a link to each distro wiki's page for is better?

zarik5 replied:
@m00nwtchr Opinions?

m00nwtchr replied:
When installing from the packages, Arch puts android sdk/ndk stuff under `/opt/`

m00nwtchr replied:
I'd keep both instructions here (for the stuff the distro wiki pages miss) and link to the distro pages.

some-username-here1 replied:
> When installing from the packages, Arch puts android sdk/ndk stuff under `/opt/` btw

So it would be as `/opt/android-sdk`?

some-username-here1 replied:
Including the above requested change, how is this?

> On Linux, the specific package names for the android tools can differ from distro to distro, see up on the wiki for more information:
> * Gentoo:
>   * https://wiki.gentoo.org/wiki/Android
> * Arch:
>   * https://wiki.archlinux.org/title/Android
> * Debian:
>   * https://wiki.debian.org/AndroidStudio
> * Ubuntu:
>   * https://help.ubuntu.com/community/AndroidSDK
> * Pop!\_OS:
>   * N/A
> 
> The three mentioned developer applications can be installed from upstream; although the packages and setup responsible for the required tools can differ between distros, being:
> 
> * **Arch**
>   * Packages can vary, read up on the Arch Wiki's [Android](https://wiki.archlinux.org/title/Android) page.
> 
> * **Gentoo**
>   * `dev-util/android-studio`
>   * `dev-util/android-sdk-update-manager`
>   * `dev-util/android-ndk >= 25.1`
> 
> For Debian, it requires to have the `non-free` repository to be enabled:
> * **Debian 12 / Ubuntu 22.10 / Pop!\_OS 22.10**
>   ```bash
>   sudo apt install android-sdk-platform-tools-common sdkmanager google-android-ndk-r25b-installer
>   ```
> 
> For Windows, set the environment variables:
> * `JAVA_HOME`:
>   * Example: `C:\Program Files\Android\Android Studio\jre`
> * `ANDROID_SDK_ROOT`:
>   * Example: `%LOCALAPPDATA%\Android\Sdk`
> * `ANDROID_NDK_HOME`:
>   * Example: `%LOCALAPPDATA%\Android\Sdk\ndk\25.1.8937393`
> 
> For Linux, the correct directories for the environment variables can greatly differ depending on the type of install. See the wiki page of your distro for more information:
> * Gentoo:
>   * https://wiki.gentoo.org/wiki/Android
> * Ubuntu:
>   * https://help.ubuntu.com/community/AndroidSDK#Post-Installation_Configuration
> 
> Distro wikis that weren't listed above does not mention of environment variables, although generally they would be as:
> * `JAVA_HOME`:
>   * `/usr/lib/jvm/default-java/bin`
> * `ANDROID_SDK_ROOT`:
>   * Arch: `~/Android/Sdk`
>   * Gentoo: `~/Android`
>   * Debian / Ubuntu / Pop!\_OS: `~/AndroidSDK`
> * `ANDROID_NDK_HOME`:
>   * Arch: `/opt/android-sdk/ndk`
>   * Linux: `/usr/lib/android-sdk/ndk`

If needed, I can add a h2 tag for each section, such as "Installing Packages" and "Environment Variables"?

some-username-here1 replied:
I have updated the building page with a few minor changes added on top of the above comment in some-username-here1/ALVR@6a34336871f0c40f24fbf77b5a9f6e7529f5b68c

How's this?
https://github.com/some-username-here1/ALVR/blob/bfs-linux-fixes/wiki/Building-From-Source.md

zarik5 commented on wiki/Building-From-Source.md:
```diff
@@ -8,6 +8,21 @@ You need to install [rustup](https://www.rust-lang.org/tools/install).
+
+Before building the server, those with nvidia GPUs would have to build and install [`FFmpeg/nv-codec-headers`](https://github.com/FFmpeg/nv-codec-headers). The nv-codec-headers requires at least the following driver version for the respective OS to work properly, taken from nv-codec-header's README:
+
+* Windows: `522.25`
```
Leave out Windows

some-username-here1 replied:
Done!

m00nwtchr commented on wiki/Building-From-Source.md:
````diff
@@ -65,14 +80,41 @@ If you want to edit and rebuild the code, you can skip the `prepare-deps` comman
+* **Arch**
+
+  ```bash
+  sudo pacman -S tandroid-sdk-platform-tools sdkmanager android-ndk
````
It's `android-sdk-platform-tools`
I'd just link to the Arch Wiki page though.

some-username-here1 replied:
What about ndk? It isn't mentioned on their wiki page, though

some-username-here1 replied:
How's this, though?

> * **Arch**
>
>    * Packages can vary, read up on the Arch Wiki's [Android](https://wiki.archlinux.org/title/Android#Other_SDK_packages_in_the_AUR) page.

m00nwtchr replied:
> What about the ndk? There's no mention of "ndk" on their wiki page, though
> 
> EDIT: Same going for Ubuntu's AndroidSDK wiki page.

Ah true, add the link but keep the install instructions then

zarik5 commented:
I'm handing off the review to @m00nwtchr since I'm not qualified enough.

some-username-here1 commented:
I have updated the rest of the wiki a bit with the text formatting and fixed the weird "Basic-" substring that appeared everywhere as a prefix to any "installation" mentioned, anything else that needs to be changed?

zarik5 commented:
There is an issue with titles with the wiki sync workflow. It seems it gets confused when the name of the page does not correspond with the topmost title within the page. Can you please go back in history and search what where the original filenames and titles of the pages? Also can you correct this in a separate PR? So here we can only discuss the building from source page.

zarik5 commented:
The wiki has been fixed by renaming the `Installation` page. Can you please rebase this PR?

some-username-here1 commented:
Undo the fixes for the other pages and placed into its own branch/PR, then rebase?

zarik5 commented:
The fix for other pages was already done. Rebase this PR just for the build from source page.

some-username-here1 commented:
Remove the `'openxr-client' Branch` from the building page, too? It has been merged a few days ago

zarik5 commented on wiki/Building-From-Source.md:
```diff
@@ -1,96 +1,156 @@
 
-To build and run:
-
+To build and run an apk for OpenXR as well, run:
```
Don't change this. You don't need to specify it's for OpenXR since it's the only client build available.

some-username-here1 replied:
Doesn't that mean that section should replace the `cargo xtask build-client --release` section?

some-username-here1 replied:
Done!

zarik5 reviewed (changes requested):
Leave an empty line between headings and paragraphs, it's a good Markdown rule.

zarik5 reviewed (changes requested):
Please also keep a new line between paragraphs and code blocks, and between bullet points. These new lines are not visible at all but they are often needed for correct parsing of the layout in some Markdown implementations.

some-username-here1 commented:
Sorry for the late response, but how are these changes?

zarik5 commented:
You don't need to put newlines between bullet point. I'm not sure if this breaks the rendering of the lists. Anyway please remove those spaces if you can do that soon or I'll merge and fix it later.

some-username-here1 commented:
Done!

zarik5 commented:
I think it's good. thanks for your patience!
